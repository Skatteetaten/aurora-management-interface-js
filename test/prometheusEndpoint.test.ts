import { request } from './helper';
import { Counter, register } from 'prom-client';

describe('Prometheus endpoint Test', () => {
  beforeEach(() => {
    register.clear();
  });

  it('should return default metrics', async () => {
    const res = await request({
      metrics: {
        enabled: true,
      },
    })
      .get('/prometheus')
      .expect(200);

    expect(res.text).not.toBe('');
  });

  it('should not return default metrics but a custom metric', async () => {
    const counter = new Counter({
      name: 'test',
      help: 'test Just a test counter',
    });

    counter.inc(2);

    const res = await request({
      metrics: {
        enabled: true,
        defaultMetrics: false,
      },
    })
      .get('/prometheus')
      .set('Accept', 'text/plain')
      .expect(200);

    expect(res.text).toMatchSnapshot();
  });
});
