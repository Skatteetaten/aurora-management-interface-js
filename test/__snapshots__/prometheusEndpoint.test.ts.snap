// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prometheus endpoint Test should not return default metrics but a custom metric 1`] = `
"# HELP test test Just a test counter
# TYPE test counter
test 2
"
`;
